packages/react-vapor/src/components/validation/hoc/WithInitialValuesMultiSelectHOC.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IMultiSelectOwnProps} from '../../select/MultiSelectConnected';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationTypes} from '../ValidationTypes';
+import {InferableComponentEnhancer} from './connectHOC';
+
+const mapDispatchToProps = (dispatch: IDispatch, {id}: IMultiSelectOwnProps) => ({
+    setWarning: (warning: string) =>
+        dispatch(ValidationActions.setWarning(id, warning, ValidationTypes.wrongInitialValue)),
+    clearWarning: () => dispatch(ValidationActions.clearWarning(id, ValidationTypes.wrongInitialValue)),
+});
+
+export type IMultiSelectWithInitialValuesOwnProps = {
+    initialValues: string[];
+    invalidInitialValuesMessage?: (invalidValues: string[]) => string;
+    resetWarningOnUnmount?: boolean;
+};
+
+const defaultInvalidInitialValuesMessageGenerator = (values: string[]) =>
+    `The initial values ('${values.join("', '")}') were selected, but not found. They will be ignored.`;
+
+export const withInitialValuesMultiSelectHOC = <T extends IMultiSelectOwnProps>(Component: React.ComponentType<T>) => {
+    type DispatchProps = ReturnType<typeof mapDispatchToProps>;
+    const WrappedMultiSelect: React.FunctionComponent<T & IMultiSelectWithInitialValuesOwnProps & DispatchProps> = ({
+        setWarning,
+        clearWarning,
+        invalidInitialValuesMessage = defaultInvalidInitialValuesMessageGenerator,
+        ...props
+    }) => {
+        const {initialValues = [] as string[], items = []} = props;
+
+        const notFoundInitialValues =
+            initialValues &&
```

Comment:
`initialValues` will always be truthy because it is set to an empty array by default on line 33. Perhaps it would be better to check the length?
```suggestion
            initialValues?.length &&
```